autoarray/inversion/mappers/delaunay.py

```diff
@@ -83,7 +83,7 @@ def __init__(
 
     @property
     def delaunay(self):
-        return self.source_pixelization_grid.Delaunay
+        return self.source_pixelization_grid.delaunay

```

Comment:
I made some pretty radical changes to `grid_2d_pixelization.py`.

Previously we had three classes that shared identical functions and properties: `Grid2DDelaunay`, `Grid2DVoronoi` and `Grid2DVoronoiNN`. This was partly my fault, but you did also copy and paste `Grid2DVoronoi` to make `Grid2DVoronoiNN`.

When classes share so many functions and properties it is best to split them such that they inherite from a common abstract parent class. This is what I've done here with the class ` AbstractGrid2DIrregular` (which I will probably rename once in a moment). This class contains all the shared methods of the 3 classes above, significantly reducing code repetition.